DB_RK3328-CC_u-boot.md

```diff
@@ -0,0 +1,141 @@
+###################### u-boot compile branch 2022.10 for roc-cc-rk3328
+
+ 
+#Dependencies
+
+sudo apt-get install gcc gcc-aarch64-linux-gnu install bc bison build-essential coccinelle   device-tree-compiler dfu-util efitools flex gdisk graphviz imagemagick   liblz4-tool libgnutls28-dev libguestfs-tools libncurses-dev   libpython3-dev libsdl2-dev libssl-dev lz4 lzma lzma-alone openssl   pkg-config python3 python3-asteval python3-coverage python3-filelock   python3-pkg-resources python3-pycryptodome python3-pyelftools   python3-pytest python3-pytest-xdist python3-sphinxcontrib.apidoc   python3-sphinx-rtd-theme python3-subunit python3-testtools   python3-virtualenv swig uuid-dev
```

Comment:
This is likely because I used Debian 11 but I'll look into it.

Reply:
Looking at these briefly I don't see any reason to keep them: 

**coccinelle**: A program matching and transformation engine that can be used to analyze and modify C code. It is often used in Linux kernel development.

**dfu-util**: A command-line utility for interacting with devices that use the USB DFU (Device Firmware Upgrade) protocol. This is often used to upgrade the firmware on embedded devices.

**gdisk**: A disk partitioning tool that supports the GPT (GUID Partition Table) partitioning scheme, which is used on modern systems with UEFI firmware.

**graphviz**: A graph visualization tool that can be used to create diagrams and charts from data in various formats.

**imagemagick**: A powerful image manipulation tool that can be used to convert, resize, and edit images in a wide variety of formats.

**liblz4-tool**: A library for compressing and decompressing data using the LZ4 compression algorithm. This library is often used in data storage and transmission applications.

**libgnutls28-dev**: A library for implementing secure communication protocols such as SSL/TLS. This library is often used in web servers, VPN software, and other network applications.

**libguestfs-tools**: A set of tools for interacting with virtual machine disk images. These tools can be used to create, modify, and inspect disk images in various formats.

**libncurses-dev**: A library for creating text-based user interfaces in a terminal. This library is often used in console applications and system utilities.

**libsdl2-dev**: A library for creating multimedia applications, including games, that run on various platforms. This library provides a simple API for working with graphics, sound, and input devices.

**lz4**: A library for compressing and decompressing data using the LZ4 compression algorithm. This library is often used in data storage and transmission applications.

**lzma & lzma-alone**: Libraries for compressing and decompressing data using the LZMA compression algorithm. These libraries are often used in data storage and transmission applications.

**openssl**: A library for implementing secure communication protocols such as SSL/TLS. This library is often used in web servers, VPN software, and other network applications.

**pkg-config**: A utility for retrieving information about installed libraries and their dependencies. This information is used by build systems to correctly link libraries when building software.

**python3-asteval**: A library for evaluating mathematical expressions in Python. This library can be used in scientific computing applications and other domains that require complex mathematical calculations.

**python3-coverage**: A library for measuring code coverage in Python programs. This library is often used in software testing and quality assurance.

**python3-filelock**: A library for creating file-based locks in Python programs. This library can be used to prevent multiple processes from accessing the same file simultaneously.

**python3-pycryptodome**: A library for implementing various cryptographic algorithms in Python. This library is often used in security applications and other domains that require secure communication and data storage.

**python3-pytest**: A testing framework for Python that allows you to write and run tests for your code. This package provides the pytest command-line tool and various plugins for extending its functionality.

**python3-pytest-xdist**: A plugin for pytest that allows you to distribute tests across multiple CPUs or machines, which can speed up test runs for large projects.

**python3-subunit**: A library for parsing and generating subunit test streams, which are a standardized format for transmitting test results between processes.

**python3-testtools**: A library of testing utilities for Python, including tools for comparing objects, mocking objects, and generating test data.

**python3-sphinxcontrib.apidoc**: A Sphinx extension for generating API documentation from Python modules. This extension can automatically generate documentation from docstrings and other metadata in your code.

**python3-sphinx-rtd-theme**: A theme for the Sphinx documentation tool that provides a modern, responsive layout optimized for reading technical documentation online.

**python3-virtualenv**: A tool for creating isolated Python environments with their own dependencies and packages. This can be useful for managing different versions of Python and their associated packages on the same system, or for creating reproducible development environments.



This was a series of recommended packages from the uboot website.

Reply:
**libgnutls28-dev** and **libssl-dev** are both libraries for implementing secure communication protocols such as SSL/TLS, but they are not interchangeable.

**libgnutls28-dev** is a library that provides an open-source implementation of TLS/SSL protocols, while **libssl-dev** is a library that provides an implementation of these protocols from the OpenSSL project. Both libraries are widely used and have their own strengths and weaknesses, and the choice of which library to use depends on the specific requirements of the software you're building.

So, if your software specifically requires the use of the GnuTLS library, then you should install **libgnutls28-dev**. If your software requires the use of the OpenSSL library, then you should install **libssl-dev**.


    **OpenSSL** is a widely used and battle-tested library, and it is the default choice for many software projects. It is known for its performance and flexibility, and it has a large community of developers and users.

    **GnuTLS** is an open-source implementation of TLS/SSL protocols that is designed to be more lightweight and portable than OpenSSL. It is also known for its strict adherence to standards and its support for newer cryptographic algorithms.


Personally I don't have a preference; I think OpenSSL is fine as it works in the Docker build you provided.

Reply:
The only dependency I was required to add with the new ubuntu jammy build was `libfontconfig1` in order to run Qt.